Modules/FindQScintilla.cmake

```diff
@@ -1,22 +1,96 @@
-# QScintilla_FOUND
-# QSCINTILLA_INCLUDE_DIR
-# QSCINTILLA_LIBRARY
-
-if(QT4_FOUND)
-  find_path(QSCINTILLA_INCLUDE_INTERNAL Qsci/qsciapi.h)
-  find_library(QSCINTILLA_LIBRARY qscintilla2)
-  if(EXISTS "{QSCINTILLA_INCLUDE_INTERNAL}" AND EXISTS "{QSCINTILLA_LIBRARY}")
-    set(QScintilla_FOUND TRUE)
-    set(SCINTILLA_INCLUDE_DIR "${SCINTILLA_INCLUDE_DIR}/Qsci")
-    set(SCINTILLA_INCLUDE_DIRS
-      "${SCINTILLA_INCLUDE_DIRS}"
-      "${QT_QTGUI_INCLUDE_DIR}"
-      "${QT_QTCORE_INCLUDE_DIR}")
-    message(STATUS "QScintilla found: ${SCINTILLA_LIBRARY}")
-  else()
-    set(QScintilla_FOUND FALSE)
+# Find QScintilla2 PyQt4 module
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# QScintilla2 website: http://www.riverbankcomputing.co.uk/software/qscintilla/
+#
+# Try to find the QScintilla2 includes and library
+# which defines
+#
+# QSCINTILLA_FOUND - system has QScintilla2
+# QSCINTILLA_INCLUDE_DIR - where to find qextscintilla.h
+# QSCINTILLA_LIBRARY - where to find the QScintilla2 library
+# QSCINTILLA_VERSION_STR - version of library
+
+# copyright (c) 2007 Thomas Moenicke thomas.moenicke@kdemail.net
+#
+# Redistribution and use is allowed according to the terms of the FreeBSD license.
+
+# Edited by Larry Shaffer, 2012
+# NOTE: include after check for Qt
+
+
+IF(QSCINTILLA_VERSION_STR)
+  # Already in cache, be silent
+  SET(QSCINTILLA_FOUND TRUE)
+ELSE(QSCINTILLA_VERSION_STR)
+
+  set(QSCINTILLA_LIBRARY_NAMES
+    qscintilla2-qt5
+    qscintilla2_qt5
+    libqt5scintilla2
+    libqscintilla2-qt5
+    qt5scintilla2
+    libqscintilla2-qt5.dylib
+    qscintilla2
```

Comment:
I had to remove this line here because on my system (Fedora 31) a QScintilla2 package exists which links against Qt4, and it gets caught by this cmake module by mistake I assume. Removing the line fixes it, could you maybe test that you can still build the project without it?

Reply:
I had just ripped off the earlier file from somewhere; wrote a cleaner (I think) clean room implementation instead which also takes care of this now.